### PR TITLE
feat(permission): add view permission control

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
Displays view objects in the application permission table ticket module

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java [hunk 1]
```diff
@@ -128,28 +133,43 @@ public List<Table> list(@NonNull @Valid QueryTableParams params) throws SQLExcep
...
+                                    .stream().map(DBObjectIdentity::getName)
+                                    .collect(Collectors.toCollection(LinkedHashSet::new));
                     generateListAndSyncDBTablesByTableType(params, database, dataSource, tables, conn,
-                            DBObjectType.EXTERNAL_TABLE, tableExtension);
+                            DBObjectType.EXTERNAL_TABLE, latestExternalTableNames);
+                }
+            }
+            if (types.contains(DBObjectType.VIEW)) {
+                ViewExtensionPoint viewExtension = SchemaPluginUtil.getViewExtension(dataSource.getDialectType());
+                if (viewExtension == null) {
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java:157]: Who will handle this exception? And is it an expected exception?

**Review by LioRoger — APPROVED** [review]
lgtm, having test case is more perfect

File: server/odc-test/src/main/java/com/oceanbase/odc/test/database/TestDBConfigurations.java [hunk 2]
```diff
@@ -150,6 +150,11 @@ private void createTestDatabasesAndUpdateConfig(TestDBType type, TestDBConfigura
                 sql = new StringBuilder("GRANT SYSDBA, RESOURCE, CREATE SESSION TO ").append(username);
                 stmt.execute(sql.toString());
                 log.info("grant sysdba to new created user, username: {}", username);
+                // Although the above code has granted sysdba role to the new user, the connection created with the
+                // new user does not use the sysdba role, so the following grant statement is required
+                sql = new StringBuilder("GRANT ALL PRIVILEGES TO ").append(username);
```
Comment (MarkPotato777) [inline server/odc-test/src/main/java/com/oceanbase/odc/test/database/TestDBConfigurations.java:155]: If grant all privileges, can we delete the above grant sql?

Comment (zijiacj) [reply server/odc-test/src/main/java/com/oceanbase/odc/test/database/TestDBConfigurations.java:155]: In fact, sysdba is granted more privileges than all privileges and is recommended to be reserved

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Review by MarkPotato777 — COMMENTED** [review]
You shoud adapt for view permission in `DBResourcePermissionHelper`, otherwise it will not work.

**Review by MarkPotato777 — CHANGES_REQUESTED** [review]
You shoud adapt for view permission in `DBResourcePermissionHelper`, otherwise it will not work.

File: server/odc-test/src/main/java/com/oceanbase/odc/test/database/TestDBConfigurations.java (hunk 2 above)
Comment (zijiacj) [reply server/odc-test/src/main/java/com/oceanbase/odc/test/database/TestDBConfigurations.java:150]: removed above code

**Comment by zijiacj** [comment]
> You shoud adapt for view permission in `DBResourcePermissionHelper`, otherwise it will not work.

have done

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Pull request merged**